firmware/arduino/chessboard/motorDriver.ino

```diff
@@ -177,22 +214,32 @@ uint8_t moveDiagonal(int startCol, int startRow, int endCol, int endRow)
   {
     setScale(xMotor, WHOLE_STEPS);
     setScale(yMotor, WHOLE_STEPS);
+    numEighthStepsX = 8;
+    numEighthStepsY = 8;
   }
   else if (numStepsY > numStepsX && (numStepsY / numStepsX) == 2)
   {
     setScale(xMotor, HALF_STEPS);
     setScale(yMotor, WHOLE_STEPS);
+    numEighthStepsX = 4;
+    numEighthStepsY = 8;
   }
   else if (numStepsY < numStepsX && (numStepsX / numStepsY) == 2)
   {
     setScale(xMotor, WHOLE_STEPS);
     setScale(yMotor, HALF_STEPS);
+    numEighthStepsX = 8;
+    numEighthStepsY = 4;
   }
   else
   {
     return INVALID_ARGS;
   }
 
+  // Sign is set based off of direction of each motor
+  numEighthStepsX *= (dirX == RIGHT) ? 1 : -1;
+  numEighthStepsY *= (dirY == UP) ? 1 : -1;
+
```

Comment:
This can be moved into the if statements, so each assignment will instead be
```c
// Sign is set based off of direction of each motor
numEighthStepsX = (dirX == RIGHT) ? MAGNITUDEX : -MAGNITUDEX;
numEighthStepsY = (dirY == UP) ? MAGNITUDEY : -MAGNITUDEY;
```
where `MAGNITUDEX` and `MAGNITUDEY` are the numbers you're assigning in each if statement

Reply:
You don't think it's better to assign magnitude then direction here? What you're suggesting would add more if statements and increase memory, but keep runtime the same.

Reply:
We're going to assign magnitude and direction instead of decomposing it to optimize the runtime slightly, so Nashir's stuff is perfect!